Mathlib/AlgebraicGeometry/EllipticCurve/Weierstrass.lean

```diff
@@ -535,7 +534,20 @@ structure EllipticCurve (R : Type u) [CommRing R] extends WeierstrassCurve R whe
 
 namespace EllipticCurve
 
-variable {R : Type u} [CommRing R] (E : EllipticCurve R)
+variable {R : Type u} [CommRing R]
+
+theorem toWeierstrassCurve_injective : Function.Injective (toWeierstrassCurve (R := R))
+  | ⟨x1, _, x3⟩, ⟨y1, _, y3⟩, h => by
+    change x1 = y1 at h
+    congr
+    exact Units.ext (by rw [x3, y3, h])
+
+@[ext]
+theorem ext {x y : EllipticCurve R} (h₁ : x.a₁ = y.a₁) (h₂ : x.a₂ = y.a₂) (h₃ : x.a₃ = y.a₃)
+    (h₄ : x.a₄ = y.a₄) (h₆ : x.a₆ = y.a₆) : x = y :=
+  toWeierstrassCurve_injective (WeierstrassCurve.ext h₁ h₂ h₃ h₄ h₆)
```

Comment:
Should it be this lemma or just `(h : x.toWeierstrassCurve = y.toWeierstrassCurve)`? In the latter case, you can tell `ext` to stop after this step and argue about bundled `WeierstrassCurve`s.

Reply:
Currently the `ext` is this one together with `x.Δ' = y.Δ'` (which is not necessary and is removed in this PR). Yours is named as `ext'` (later renamed to `toWeierstrassCurve_injective`) in this PR. I don't know which one is better for the ext lemma, though.

Reply:
I've checked that
```lean
@[ext]
theorem ext : Function.Injective (toWeierstrassCurve (R := R))
```
works and the later proof using `ext` is not affected. Is this a good idea?

Reply:
I think the current version is better.

Reply:
As in: I think the PR is LGTM as it is.